docs/repo-guides/release-nnf-sw/readme.md

```diff
@@ -89,93 +91,101 @@ just an example.
         determine which commits have been merged, and then when the next release is created Git will
         treat everything like a conflict. Additionally, this will cause auto-generated release notes
         to include the previous release.
 
-7. Once merged, update the release branch locally and then create an annotated tag:
+7. Once merged, update the release branch locally and create an annotated tag:
 
     ```shell
     git checkout releases/v0
+    git pull
     git tag -a v0.0.3 -m "Release v0.0.3"
     git push origin --tags
     ```
 
-8. Now that there is a tag, a release can be created via the [GitHub CLI](https://cli.github.com/).
+8. Now that a tag exists, a release can be created via the [GitHub CLI](https://cli.github.com/).
    Alternatively, use the [Web UI](https://github.com/NearNodeFlash/nnf-dm/releases/new).
 
     ```bash
     gh release create --generate-notes --verify-tag -p v0.0.3 -t "Release v0.0.3"
     ```
 
-9. Repeat this process for each remaining component.
+9. GOTO Step 1 and repeat this process for each remaining component.
 
 ## Release `nnf-deploy`
 
-Once the individual components are released, we need to update the submodules and
-`config/repositories.yaml` in the **master** branch before we start on the release branch. This
-makes sure that everything is now current on master.
+Once the individual components are released, we need to update the submodules
+in the `master` branch before we start on the release branch. This ensures
+that everything is current on `master` for `nnf-deploy`.
 
-1. Update the submodules on master:
+1. Update the submodules for `nnf-deploy` on master:
 
     ```shell
     git checkout master
     git pull
-    ./update.sh
     ```
 
-2. Update `config/repositories.yaml` and update the referenced versions for:
-
-    1. `lustre-csi-driver`
-
-    2. `lustre-fs-operator`
+2. Create a branch to capture the submodule changes for the PR to `master`
 
-    3. `nnf-mfu`
+    ```shell
+    git checkout -b update-submodules
+    ./update.sh
+    ```
 
 3. Commit the changes and open a Pull Request against the `master` branch.
 
-4. Once merged, follow steps 1-3 from the previous section to update the release branch with master.
+4. Once merged, follow steps 1-3 from the previous section to create a release branch off of `releases/v0` and
+   update it with changes from `master`.
 
-5. There will be conflicts on the submodules after step 3. This is expected. We will update the
+5. There will be conflicts for the submodules after step 3. **This is expected.** Update the
    submodules to the new tags and then commit the changes.  If each tag was committed properly, the
    following command can do this for you:
 
     ```shell
-    git submodule foreach 'git checkout `git describe --match="v*" HEAD`'
+    git submodule foreach 'git checkout $(git describe --tags --abbrev=0 $(git rev-list --tags --max-count=1))'
```

Comment:
I think we should put this in a script and keep it in nnf-deploy rather than rely on this documenation for this one.